### ISSUE TITLE
UX: improve error message when test is filtered out

### ISSUE THREAD

**Issue by melix** [gradle/gradle#16029]
### Context

This caused me a headache which has hopefully been mitigated thanks to the help of @rieske . Basically, if a test task is configured to filter out tests:

```
    register<SmokeTest>("smokeTest") {
        description = "Runs Smoke tests"
        configureForSmokeTest(*santaProjects)
        useJUnitPlatform {
            filter {
                excludeTestsMatching(gradleBuildTestPattern)
            }
        }
    }
```

and that you run via CLI, or in my case, via the IDE, this test task, you get this error message:

```
> No tests found for given includes: [org.gradle.smoketests.GradleBuildConfigurationCacheSmokeTest](filter.includeTestsMatching)
```

which is not quite correct: tests **were** found, but they were **filtered out**. So it's not a bug in the engine not finding the test, it's actually that I was calling the wrong task.

Suggestion:

```
> No tests were executed for given includes: [org.gradle.smoketests.GradleBuildConfigurationCacheSmokeTest](filter.includeTestsMatching) but the following tests were found and ignored by a filter on the task configuration:
   - org.gradle.smoketests.GradleBuildConfigurationCacheSmokeTest
```

**Comment by ldaley** [comment]
The experience is similar when using `--tests`. The feedback is close to incomprehensible.

**Comment by melix** [comment]
Yes, this is the very same issue: when you run a test from IDEA like I did, it actually uses `--tests` and the fact Gradle tells you that the test you just started doesn't exist is... frustrating.

**Comment by melix** [comment]
ping

**Comment by sschuberth** [comment]
Actually, I believe the "No tests found for given includes" message is fine. It all boils down to who you think "found" refers to: To the test engine, or to Gradle. To me it refers to Gradle, as the message comes from Gradle. So Gradle did not "find" the test (because it was filtered), but the test engine did find it.

Anyway, what I believe is more confusing than that is that the message talks about "includes" although only an exclude was defined. Maybe better use a wording like "No tests to execute remaining after filtering".

**Comment by lingocoder** [comment]
> …_The experience is similar when using `--tests`. The feedback is close to incomprehensible_…

[_Can confirm_](https://github.com/gradle/gradle/issues/26647) :(
